includes/menu.inc

```diff
@@ -303,3 +303,66 @@ function islandora_batch_set_restart_batch_form_submit(&$form, &$form_state) {
     'ingest_set' => $id,
   ));
 }
+
+/**
+ * Page callback to purge ingested objects that belong to a set.
+ */
+function islandora_batch_purge_set_object_page_callback($id, $management = FALSE) {
+  return drupal_get_form('islandora_batch_purge_set_object_form', $id, $management);
+}
+
+/**
+ * Form building function to purge ingested objects.
+ */
+function islandora_batch_purge_set_object_form($form, &$form_state, $id, $management) {
+  form_load_include($form_state, 'inc', 'islandora_batch', 'includes/ingest.batch');
+
+
+  $lock_name = islandora_batch_get_lock_name($id);
+
+  $form['info'] = array(
+    '#type' => 'item',
+    '#markup' => t('This will attempt to purge ingested objects in set @set from the repository.', array(
+      '@set' => $id,
+    )),
+  );
+
+  $form['global_notice'] = array(
+    '#type' => 'item',
+    '#title' => t('Global batch ingest lock'),
+    '#markup' => lock_may_be_available(ISLANDORA_BATCH_LOCK_NAME) ? t('Available') : t('Locked'),
+    '#description' => t('If this is locked, there may be a general batch being run over all items in the queue (in which case we will have to wait until it completes), or it may be possible that another user has just started a batch (in which case reloading the page may show that it is now available).'),
+  );
+
+  $set_lock = lock_may_be_available($lock_name);
+  $form['set_notice'] = array(
+    '#type' => 'item',
+    '#title' => t('Set batch ingest lock'),
+    '#markup' => $set_lock ? t('Available') : t('Locked'),
+    '#description' => t('The lock for the current set, to prevent other entities from trying to process the same set of items concurrently.'),
+  );
+
+  $form['lock_info'] = array(
+    '#type' => 'item',
+    '#markup' => t('If the lock is not avaliable, the purge process will attempt to obtain one before proceeding.'),
+  );
+
+
+  return confirm_form(
+    $form,
+    t('Purge ingested objects in set "@set"?', array(
```

Comment:
Could we add a more verbose confirmation text here, like "This will definitively purge ...(same as currently) + from your repository? Even when using the word Purge, it could be not clear enough to people that they are going to "effectively" remove Islandora Objects (and not only preprocess objects from the set) which could mean, that if badly used, could cost someone's job 😄